custom/covid/management/commands/update_owner_ids.py

```diff
@@ -0,0 +1,55 @@
+from xml.etree import cElementTree as ElementTree
+
+from casexml.apps.case.mock import CaseBlock
+from dimagi.utils.chunked import chunked
+
+from corehq.apps.hqcase.utils import submit_case_blocks
+from corehq.form_processor.interfaces.dbaccessors import CaseAccessors
+from corehq.apps.locations.models import SQLLocation
+from custom.covid.management.commands.update_cases import CaseUpdateCommand
+
+BATCH_SIZE = 100
+DEVICE_ID = __name__ + ".update_owner_ids"
+CHILD_LOCATION_TYPE = 'investigators'
+
+
+class Command(CaseUpdateCommand):
+    help = f"Changes the owner_id of a case to the location_id of the child location with type " \
+           f"{CHILD_LOCATION_TYPE} of the current location"
+
+    def case_block(self, case, child_location):
+        return ElementTree.tostring(CaseBlock.deprecated_init(
+            create=False,
+            case_id=case.case_id,
+            owner_id=child_location.location_id,
+        ).as_xml()).decode('utf-8')
+
+    def update_cases(self, domain, case_type, user_id):
+        case_ids = self.find_case_ids_by_type(domain, case_type)
+        accessor = CaseAccessors(domain)
+
+        case_blocks = []
+        skip_count = 0
+        for case in accessor.iter_cases(case_ids):
+            owner_id = case.get_case_property('owner_id')
+            location_obj = SQLLocation.objects.get(location_id=owner_id)
```

Comment:
Not blocking: Since all of the cases will be owned by a small number of locations, it'd save some resources to not do this lookup for each case, but instead to save a little cache of location objects and then only hit the database if the current case's owner isn't already there.

Reply:
Good call [cb9ef92](https://github.com/dimagi/commcare-hq/pull/29011/commits/cb9ef92328788bc5f66e28bbf2dd33d33c56cacb)